packages/workbox-precaching/controllers/PrecacheController.mjs

```diff
@@ -301,6 +301,20 @@ class PrecacheController {
       };
     });
   }
+
+  /**
+   * Returns an array of fully qualified URL's that will be precached.
+   *
+   * @return {Array<string>} An array of URLs.
+   */
+  async getCachedUrls() {
```

Comment:
I'm thinking that it would be safer to always go through the steps of creating a fresh array of URLs and not saving and reusing the previous `this._parsedCacheUrls` value, since the underlying `this._entriesToCacheMap` value might be changed at any time.

Alternatively, you could set `this._parsedCacheUrls` to `null` whenever `this._entriesToCacheMap` is modified, but that's probably overkill.

(This might have been an issue with v2, too.)